tests/atest/rules/naming/possible-variable-overwriting/test.robot

```diff
@@ -0,0 +1,27 @@
+*** Test Cases ***
+Test With Overwritten Variables
+    ${my_var}       Set Variable    1
+    ${my_var}       Set Variable    2
+    ${MY VAR}       Set Variable    2
+    ${myVar}        Set Variable    3
+    ${another_var}  Set Variable    4
+    ${MYVAR}        Set Variable    5
+    ${M Y V A R}    Set Variable    6
+    ${mYvAr}        Some Keyword    123
+
+
+*** Keywords ***
+Keyword With Overwritten Variables
+    [Arguments]     ${my_var}
+    ${someVar}      Set Variable    1
+    ${SOME_VAR}     Set Variable    2
```

Comment:
Add more testing with:
- for loops (especially if nothing breaks and nested for loops are visited)
- if conditions
- Templated tests (the ones with [Template])